scipy/optimize/_linprog_util.py

```diff
@@ -1424,28 +1432,18 @@ def _check_result(x, fun, status, slack, con, bounds, tol, message):
         message = ("The solution does not satisfy the constraints within the "
                    "required tolerance of " + "{:.2E}".format(tol) + ", yet "
                    "no errors were raised and there is no certificate of "
-                   "infeasibility or unboundedness. This is known to occur "
-                   "if the `presolve` option is False and the problem is "
-                   "infeasible. This can also occur due to the limited "
-                   "accuracy of the `interior-point` method. Check whether "
+                   "infeasibility or unboundedness. Check whether "
                    "the slack and constraint residuals are acceptable; "
-                   "if not, consider enabling presolve, reducing option "
-                   "`tol`, and/or using method `revised simplex`. "
-                   "If you encounter this message under different "
-                   "circumstances, please submit a bug report.")
-    elif status == 0 and contains_nans:
-        status = 4
-        message = ("Numerical difficulties were encountered but no errors "
-                   "were raised. This is known to occur if the 'presolve' "
-                   "option is False, 'sparse' is True, and A_eq includes "
-                   "redundant rows. If you encounter this under different "
-                   "circumstances, please submit a bug report. Otherwise, "
-                   "remove linearly dependent equations from your equality "
-                   "constraints or enable presolve.")
+                   "if not, consider enabling presolve, adjusting the "
+                   "tolerance option(s), and/or using a different method. "
+                   "Please consider submitting a bug report.")
     elif status == 2 and is_feasible:
         # Occurs if the simplex method exits after phase one with a very
         # nearly basic feasible solution. Postsolving can make the solution
         # basic, however, this solution is NOT optimal
-        raise ValueError(message)
```

Comment:
I don't know when this got added. It doesn't make sense to me for `linprog` to raise an error rather than a nonzero status, so I'm changing this while it's on my radar.